packages/manager/modules/logs-to-customer/src/data/hooks/useLogTailUrl.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from '@tanstack/react-query';
+import { postLogTailUrlv2, postLogTailUrlv6 } from '../api/logTailUrl';
+import { LogApiVersion } from '../types/apiVersion';
+import { ApiUrls } from '../../LogsToCustomer.module';
+
+interface IUseLogTailUrl {
+  logTailUrl: ApiUrls['logUrl'];
+  logKind: string;
+  apiVersion: LogApiVersion;
+}
+
+export const getLogTailUrlQueryKey = ({
+  logTailUrl,
+  logKind,
+}: Omit<IUseLogTailUrl, 'apiVersion'>) => [`postLogUrl`, logTailUrl, logKind];
+
+export const useLogTailUrl = ({
+  logTailUrl,
+  logKind,
+  apiVersion,
+}: IUseLogTailUrl) => {
```

Comment:
Using an object may be overkill passing it as param would be sufficient i think ?